src/app/components/tables/data-table/terra-data-table-base.service.ts

```diff
@@ -0,0 +1,108 @@
+import { TerraBaseService } from '../../../service/terra-base.service';
+import { Observable } from 'rxjs/Observable';
+import { TerraPagerParameterInterface } from '../../pager/data/terra-pager.parameter.interface';
+import { TerraPagerInterface } from '../../pager/data/terra-pager.interface';
+import { TerraSelectBoxValueInterface } from '../../forms/select-box/data/terra-select-box.interface';
+import { Http } from '@angular/http';
+import { TerraLoadingSpinnerService } from '../../loading-spinner/service/terra-loading-spinner.service';
+import { TerraDataTableSortOrder } from './terra-data-table-sort-order.enum';
+
+/**
+ * @author pweyrich
+ */
+export abstract class TerraDataTableBaseService<T, P> extends TerraBaseService
+{
+    public requestPending:boolean;
+    public onSuccessFunction:(res:Array<T>) => void;
+    public pagingData:TerraPagerInterface<T>;
+    public pagingSizes:Array<TerraSelectBoxValueInterface>;
+    public defaultPagingSize:number;
+    public filterParameter:P;
+    public sortBy:string;
+    public sortOrder:TerraDataTableSortOrder;
+
+    constructor(loadingSpinner:TerraLoadingSpinnerService, http:Http)
+    {
+        super(loadingSpinner, http, '');
+    }
+
+    /**
+     * @description Updates the stored paging data with the given data
+     * @param {TerraPagerInterface} pagerData
+     */
+    public updatePagingData(pagerData:TerraPagerInterface<T>):void
+    {
+        this.pagingData = {
+            page:           pagerData.page,
+            itemsPerPage:   pagerData.itemsPerPage,
+            totalsCount:    pagerData.totalsCount,
+            isLastPage:     pagerData.isLastPage,
+            lastPageNumber: pagerData.lastPageNumber,
+            firstOnPage:    pagerData.firstOnPage,
+            lastOnPage:     pagerData.lastOnPage
+        };
+    }
+
+    /**
+     * @description Wrapper for the abstract requestTableData method. All the default behaviour when retrieving data is implemented here.
+     * @param fromFilter
+     */
+    public getResults(fromFilter?:boolean):void
```

Comment:
```suggestion
    public getResults(fromFilter:boolean = false):void
```

Damit es korrekt gesetzt wird